### ISSUE TITLE
permission on /var/run/php5-fpm.sock problem

### ISSUE THREAD

**Issue by hsum** [stephenliang/mediawiki-dockerfile#2]
The wiki doesn't come up and I get this error:

2015/03/17 09:01:28 [crit] 11#0: *1 connect() to unix:/var/run/php5-fpm.sock fai
led (13: Permission denied) while connecting to upstream, client: 172.17.42.1, s
erver: _, request: "GET / HTTP/1.1", upstream: "fastcgi://unix:/var/run/php5-fpm
.sock:", host: "localhost:7676"

I found changing the group to www-data allows you to get past this:
root@c15aa7b132a8:/src# chgrp www-data /var/run/php5-fpm.\*  

Unfortunately, the socket gets recreated as root.root after restarting.

Do you know how to fix this?

thanks.

**Comment by stephenliang** [comment]
That's very interesting, what distro are you running? It seems that there may be a SELinux/AppArmor issue. I run these guys on Ubuntu 14.04/14.10 servers and they run fine.

Speaking of which, I just ran MediaWiki and checked `/var/run`, I see they are spawning with www-data:

```
    root@4b4825ac04bc:/# ls -l /var/run/php5-fpm.sock
    srw-rw---- 1 www-data www-data 0 Mar 18 01:15 /var/run/php5-fpm.sock
```

**Comment by hsum** [comment]
Thank you for your insight.  I'm running Ubuntu 14.04.

I got rid of apparmor on my host, not in container:
service apparmor stop
update-rc.d -f apparmor remove

then generated the LocalSettings.php.  It worked as expected after that:

root@0ba487767ab8:/# ls -ltr /var/run/php5-fpm.sock
srw-rw---- 1 www-data www-data 0 Mar 18 02:16 /var/run/php5-fpm.sock
1. I provided the ip address of the mysql container obtained via inspect.  Is there a more hands-free way?  Perhaps a way to generate LocalSettings.php without using the web browser?
2. Can apparmor be disabled as part of the Dockerfile or does apparmor need to be removed on the host?

thanks.
